### DISCUSSION TITLE
Unlock access to domain with custom filtering rule

### DISCUSSION THREAD

**Discussion by lmangone** [AdguardTeam/AdGuardHome#3319] [Q&A]
Start of discussion in the forum : https://forum.adguard.com/index.php?threads/adguard-home-filtering-setup-question.43674/#post-212304

### Issue Details

* **Version of AdGuard Home server:**
  * v0.106.2
* **How did you install AdGuard Home:**
  * Installation from docker hub website
* **How did you setup DNS configuration:**
  * router primary DNS setup with adguard server IP address
* **CPU architecture:**
  * intel
* **Operating system and version:**
  * Debian buster with OMV that runs adguard docker image

### Expected Behavior

Unlock web site filtering with the following custom rule "@@||animedigitalnetwork.fr^$important"

### Actual Behavior

Web site detect Ads blocker despite custom filtering rule

### Screenshots

<details><summary>Screenshot: 
</summary>

![image](https://user-images.githubusercontent.com/17402030/124498688-0c4f1980-ddbd-11eb-884d-947bdb3ca58f.png)
</details>

### Additional Information

I tried to clear my windows laptop dns cache with `ipconfig /flushdns` command and logoff login to my wifi but has no effect. Android devices has the same problem.

**Comment by ainar-g** [comment]
Hello.  We cannot reproduce the issue.  Have you also cleared AGH's cache?  Try setting it to zero and then resetting it to previous value in “DNS settings” → “DNS cache configuration”.

**Comment by lmangone** [comment]
I set no value to the cache size and put a TTL max value=60s but it as still no effect when activating the AGH protection.
What is new for me is that when I disable AGH protection, It works fine (was not the case before).

I also tried the following custom filtering rules
```
@@||https://animedigitalnetwork.fr^$important
@@||animedigitalnetwork.fr^$important
```

**Comment by ameshkov** [comment]
Could you please examine the query log, what exact rule is blocking `animedigitalnetwork.fr`?

**Comment by lmangone** [comment]
@ameshkov where can I get this information ? From AGH logs or from web browser developper tools ?

**Comment by lmangone** [comment]
Here are the custom filter rules I put in AGH
```
@@||https://animedigitalnetwork.fr^$important
@@||animedigitalnetwork.fr^$important
```
No DNS cache size and a TTLmax=60s

Here are the Chrome console logs
<html>
<body>
<div tabindex="-1" class="console-message-wrapper console-warning-level" style="box-sizing: border-box; min-width: 0px; min-height: 0px; display: flex; border-top-width: ; border-top-style: ; border-top-color: var(--warning-border-color); border-bottom: 1px solid transparent; background-color: rgb(255, 251, 229); color: rgb(34, 34, 34); font-family: consolas, &quot;lucida console&quot;, &quot;courier new&quot;, monospace; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="console-message" style="box-sizing: border-box; min-width: 0px; min-height: 17px; clear: right; position: relative; padding: 3px 22px 1px 0px; margin-left: 24px; flex: 1 1 auto; display: flex;"><span class="source-code" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size)  !important; flex: 1 1 auto; line-height: 1.2;"><span class="console-message-anchor" style="box-sizing: border-box; min-width: 0px; min-height: 0px; float: right; text-align: right; max-width: 100%; margin-left: 4px;"><span class="devtools-link" role="link" tabindex="-1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: rgb(84, 84, 84); text-decoration: underline; cursor: pointer; word-break: break-all;">2.b7095c03.chunk.js:1</span> </span><span class="console-message-text" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: rgb(92, 60, 0);">Invalid asm.js: Unexpected token</span></span></div></div><div tabindex="-1" class="console-message-wrapper console-error-level console-selected" style="box-sizing: border-box; min-width: 0px; min-height: 0px; outline-width: 0px; display: flex; border-top-width: ; border-top-style: ; border-top-color: var(--error-border-color); border-bottom: 1px solid rgb(204, 222, 245); background-color: rgb(236, 241, 248); --error-text-color: rgb(200 0 0); color: rgb(34, 34, 34); font-family: consolas, &quot;lucida console&quot;, &quot;courier new&quot;, monospace; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="console-message" style="box-sizing: border-box; min-width: 0px; min-height: 17px; clear: right; position: relative; padding: 3px 22px 1px 0px; margin-left: 24px; flex: 1 1 auto; display: flex;"><span is="ui-icon" class="message-level-icon spritesheet-smallicons smallicon-error" aria-label="Error" style="box-sizing: border-box; min-width: 0px; min-height: 0px; display: inline-block; flex: 1 1 auto; position: absolute; left: -17px; top: 4px; user-select: none; background-position: var(--spritesheet-position); background-image: var(--image-file-smallIcons); --spritesheet-position:-40px 70px; width: 10px; height: 10px;"></span><div class="console-message-stack-trace-toggle" style="box-sizing: border-box; min-width: 0px; min-height: 0px; flex: 1 1 auto; display: flex; flex-direction: row; align-items: flex-start;"><div class="console-message-stack-trace-wrapper" role="tree" style="box-sizing: border-box; min-width: 0px; min-height: 0px; flex: 1 1 auto; display: flex; flex-direction: column; align-items: stretch;"><div role="treeitem" aria-expanded="true" tabindex="-1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; flex: 0 0 auto;"><span is="ui-icon" class="console-message-expand-icon spritesheet-smallicons smallicon-triangle-down icon-mask" style="box-sizing: border-box; min-width: 0px; min-height: 0px; display: inline-block; flex-shrink: 0; margin-bottom: -2px; background-color: rgb(110, 110, 110); -webkit-mask-position: var(--spritesheet-position); -webkit-mask-image: var(--image-file-smallIcons); --spritesheet-position:-80px 30px; width: 10px; height: 10px;"></span><span class="source-code" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size)  !important; line-height: 1.2;"><span class="console-message-anchor" style="box-sizing: border-box; min-width: 0px; min-height: 0px; float: right; text-align: right; max-width: 100%; margin-left: 4px;"><span class="devtools-link" role="link" tabindex="-1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: rgb(84, 84, 84); text-decoration: underline; cursor: pointer; word-break: break-all;">Helmet.js:493</span> </span><span class="console-message-text" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: var(--error-text-color)  !important;">GET <span class="devtools-link" role="link" tabindex="-1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: rgb(84, 84, 84); text-decoration: underline; cursor: pointer; word-break: break-all;">https://securepubads.g.doubleclick.net/tag/js/gpt.js</span> net::ERR_NAME_NOT_RESOLVED</span></span></div><div class="" role="group" style="box-sizing: border-box; min-width: 0px; min-height: 0px; flex: 0 0 auto;"><span class="monospace" style="display: inline-block; --monospace-font-size: 12px; --monospace-font-family: consolas, lucida console, courier new, monospace; --source-code-font-size: 12px; --source-code-font-family: consolas, lucida console, courier new, monospace; box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--monospace-font-family); font-size: var(--monospace-font-size)  !important;">

  | (anonymous) | @ | Helmet.js:493
-- | -- | -- | --
  | oe | @ | Helmet.js:492
  | te | @ | Helmet.js:372
  | (anonymous) | @ | Helmet.js:342
  | requestAnimationFrame (async) |   |  
  | (anonymous) | @ | Helmet.js:341
  | c | @ | index.js:65
  | a.UNSAFE_componentWillMount | @ | index.js:99
  | wo | @ | react-dom.production.min.js:3202
  | Ba | @ | react-dom.production.min.js:4307
  | gu | @ | react-dom.production.min.js:6700
  | ls | @ | react-dom.production.min.js:6150
  | us | @ | react-dom.production.min.js:6139
  | $u | @ | react-dom.production.min.js:5880
  | Hu | @ | react-dom.production.min.js:5678
  | Ls | @ | react-dom.production.min.js:7086
  | (anonymous) | @ | react-dom.production.min.js:7181
  | ts | @ | react-dom.production.min.js:5931
  | Ws | @ | react-dom.production.min.js:7180
  | t.render | @ | react-dom.production.min.js:7363
  | 855 | @ | index.js:13
  | s | @ | 10210-episode-137-l-…ays-des-samourais:4
  | 371 | @ | main.2bcc0f3c.chunk.js:1
  | s | @ | 10210-episode-137-l-…ays-des-samourais:4
  | l | @ | 10210-episode-137-l-…ays-des-samourais:4
  | e | @ | 10210-episode-137-l-…ays-des-samourais:4
  | (anonymous) | @ | main.2bcc0f3c.chunk.js:1

</span></div></div></div></div></div>
</body>
</html>

**Reply by ameshkov** [reply]
You should simply examine AdGuard Home's query log and see what's blocked.

`@@||animedigitalnetwork.fr^$important` - this rule unblocks `animedigitalnetwork.fr`, it does not unblock sub-requests sent from the pages of that website.

**Reply by lmangone** [reply]
Got it, I'll check if I can unblock sub-requests too to have something works as intended.

**Reply by lmangone** [reply]
Thanks @ameshkov, I manage to make it work by unblocking some DNS from the query logs.
Best alternive I've found so far would have been to unblock everything for a specific client (my TV) : @@||*^$client=192.168.0.12

Is there any more convienient way to unblock a specific web site for all devices ?

**Reply by ameshkov** [reply]
Alternatively, you could add your TV as a "Client" and uncheck "Block domains" there
